src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -110,13 +107,19 @@ def _write_to_file(file_path, content):
         file.write(content)
 
 
-def _fetch_image(registry, image_name, job_dir, tag='latest', force_pull=True):
+def _fetch_image(registry, tool, job_dir, force_pull=True):
     """
     Fetches the specified Shifter image if it is not already present on the system.
 
     When force_pull is set to True, the image is always pulled from the registry
     """
-    image_str = f"{registry}/{image_name}:{tag}"
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    compute_tools_dir = os.path.join(current_dir, COMPUTE_TOOLS_DIR)
+    version_file = os.path.join(compute_tools_dir, tool, VERSION_FILE)
+    tool_img_ver = extract_latest_version(version_file)
+    tool_img_tag = f'{tool}_{tool_img_ver}'
+    image_str = f'{registry}:{tool_img_tag}'
 
     if not force_pull:
```

Comment:
It seems like this code is never executed since `force_pull` is always `True`. To my mind this code should always be executed, right?

It also seems like the force_pull arg could be removed since it's always the same, unless there's some reason to keep it I'm not seeing.

Reply:
removed unused codes. I think we should always run `shifterimg pull`.

Reply:
ok. reversed changes.

Reply:
I'd still like to understand why you thought we should pull the image every time, maybe I'm missing something

Reply:
I just thought it would be safer to pull every time. And this operation is not very time consuming.

Reply:
Why would it be safer? (I'm not trying to be argumentative here, just in case I'm coming off that way, genuinely want to understand) 

If it's cheap then maybe pulling out all that code is the right way to go

Reply:
I think it's safer because it's pulling every time.

Reply:
right, but at the risk of sounding like a broken record, why is it safer to pull every time if the image has a specific tag that doesn't change?

Reply:
Yea. I think what you said makes sense. We are using a specific tag.

Reply:
Fair enough